### PR TITLE
Adding recaptcha front end.

### PR DESCRIPTION
**Pull request by ScottDowne**
WIP I left some notes for things we can still fix.

I havn't done back end yet.

I need to add stuff so it doesn't bork via review app or tests. Basically, turn it on and off when it's production or explicitly turned on locally to test.

### CODE REVIEW

File: src/mixins/stripe.js [hunk 1]
```diff
@@ -64,6 +64,14 @@ var StripeMixin = {
       label: error
     });
   },
+  reCaptcha: function(callback) {
+    grecaptcha.render('g-recaptcha', {
```
Comment (ScottDowne) [inline src/mixins/stripe.js]: So thinking about putting the content of this function inside of some check to make sure `grecaptcha` is on the window, this way we can lazy load the captcha script while the user interacts with the form itself, likely being ready by the time they are done.

File: src/mixins/stripe.js [hunk 2]
```diff
@@ -64,6 +64,14 @@ var StripeMixin = {
...
+      'sitekey' : '6LeVaUwUAAAAAM6TtwWfLQLXEWY5sSzEYdWT3fYK',
```
Comment (ScottDowne) [inline src/mixins/stripe.js]: Make this an .env

File: server/routes/index.js [hunk 3]
```diff
@@ -70,6 +76,32 @@ const mailchimp = async function(request, h) {
 const routes = {
   signup: require('./signup'),
   mailchimp,
+  'reCaptcha': async function(request, h) {
+    const payload = request.payload || {};
+    const reCaptchaToken = payload.reCaptchaToken || "";
```
Comment (cadecairos) [inline server/routes/index.js]: with Joi requiring this to be set, does it make sense to even have the `|| "";`

File: src/mixins/stripe.js (hunk 1 above)
Comment (cadecairos) [reply src/mixins/stripe.js]: would that be difficult?

File: server/routes/index.js (hunk 3 above)
Comment (ScottDowne) [reply server/routes/index.js]: Probably not. I guess if the joi stuff ever gets refactored, this'll fail a bit more gracefully?

File: app.json [hunk 4]
```diff
@@ -84,6 +84,12 @@
     "PAYPAL_USER": {
       "required": true
     },
+    "RECAPTCHA_SECRET_KEY": {
+      "required": true
```
Comment (cadecairos) [inline app.json]: Lets make these the test keys

**Comment by ScottDowne** [comment]
Found a bug while banging against this. If you close stripe an reopen it, recaptcha fails saying it's already been initialized.

File: server/routes/index.js [hunk 5]
```diff
@@ -81,6 +80,9 @@ const routes = {
     const reCaptchaToken = payload.reCaptchaToken || "";
 
     return new Promise((resolve, reject) => {
+      if (process.env.RECAPTCHA_DISABLED) {
+        return resolve("OK");
```
[the remark this answers is not in the document]
Comment (ScottDowne) [reply server/routes/index.js]: Good idea.

**Comment by ScottDowne** [comment]
Getting this on staging now :rocket:

**Pull request merged**